usecases/tasks_handler.go

```diff
@@ -0,0 +1,251 @@
+	// CreateTask(task *entities.Task) error
+	// GetAllTaskByStatus(taskType int32, taskStatus int32) ([]*entities.Task, error)
+	// UpdateTasksStatus(taskIds []int64, taskStatus int32) error
+	// UpdateTaskStatus(taskId int64, taskStatus int32) error
+	// FindOneById(taskId int64) (*entities.Task, error)
+	// GetRawTransactionByBlockNumber(blockNumber uint64) ([]*entities.RawTransaction, error)
+}
+
+type taskHandler struct {
+	dbRepo dbRepo
+	logger *flogging.FabricLogger
+}
+
+func NewTaskHandler(dbRepo dbRepo) *taskHandler {
+	return &taskHandler{
+		dbRepo: dbRepo,
+		logger: flogging.MustGetLogger("corev4-explorer.usecase.task.handler"),
+	}
+}
+
+func (t *taskHandler) PerformTasks(taskId int64, tasksType int32) common.BaseError {
+	switch tasksType {
+	case int32(entities.TaskTransform):
+		return t.transformData(taskId)
+	case int32(entities.TaskBalance):
+		return t.calculateAccountDelta(taskId)
+	default:
+		return common.NewUnknownError(errors.New("unknown task"))
+	}
+}
+
+func (t *taskHandler) transformData(taskId int64) common.BaseError {
+	t.logger.Infof("Transform task %d start", taskId)
+	rawTxs, err := t.dbRepo.QueryRawTxsByTaskTransformID(taskId)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+
+	var onchainTxList []*entities.OnchainTransaction
+
+	var accountList []*entities.Account
+
+	var accountTxs []*entities.AccountTx
+
+	for _, rawTransaction := range rawTxs {
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Transaction.") {
+			newTransactions, newAccountTxs, err := t.rawTxToTransactions(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			onchainTxList = append(onchainTxList, newTransactions...)
+			accountTxs = append(accountTxs, newAccountTxs...)
+
+		}
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Account.") {
+			newAccounts, err := t.rawTxToAccounts(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			accountList = append(accountList, newAccounts...)
+
+		}
+
+	}
+	err = t.dbRepo.SaveTransformData(taskId, onchainTxList, accountList, accountTxs)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+	return nil
+}
+
+func (t *taskHandler) calculateAccountDelta(taskId int64) common.BaseError {
+	return nil
+}
+
+func (t *taskHandler) rawTxToTransactions(rawTransaction *entities.RawTransaction, taskId int64) ([]*entities.OnchainTransaction, []*entities.AccountTx, common.BaseError) {
+	var onchainTxList []*entities.OnchainTransaction
+	var accountTxsList []*entities.AccountTx
+
+	response, err := contractpb.NewTransactionResponse(rawTransaction.Payload)
+	if err != nil {
+		t.logger.Errorf("Create new transaction response error %s", err.Error())
+		return nil, nil, common.NewUnknownError(err)
+	}
+
+	for _, attribute := range response.Transaction.Attributes {
+
+		preDay, err := time.Parse(common.YYYYMMDD, attribute.OffchainTime.PreDay)
```

Comment:
Ensure that errors during time parsing in `rawTxToTransactions` are returned instead of just being logged. This will prevent silent failures and improve error visibility. [important]